library/std/src/time.rs

```diff
@@ -125,6 +124,26 @@ pub use core::time::FromSecsError;
 /// > structure cannot represent the new point in time.
 ///
 /// [`add`]: Instant::add
+///
+/// ## Monotonicity
+///
+/// All [tier 1] and most tier 2 platforms with std support rely on monotonicity guarantees
+/// from the operating system. Rarely these guarantees are broken by hardware, virtualization
+/// or operating system bugs. In previous versions this resulted in panics in
+/// [`duration_since`], [`elapsed`] and [`sub`]. Currently they will either saturate to zero or,
+/// to let tests detect logic bugs such as swapped earlier and later times, only panic in debug mode.
```

Comment:
```suggestion
/// to let tests detect logic bugs such as swapped earlier and later times, panic when overflow checks are enabled.
```

Right? Debug mode is not really describing this perfectly.